docs/running-recipes/popular-recipe-guides/migrate-to-java-25.md

```diff
@@ -0,0 +1,232 @@
+---
+keywords: [OpenRewrite, Java, Java 25, Java Migration]
+description: How to automatically migrate from Java 21 to 25.
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Migrate to Java 25
+
+In this tutorial, we'll use OpenRewrite to perform an automated migration from Java 21 to Java 25. While this recipe _will_ upgrade your Java 8, 11, or 17 projects to Java 25, this guide won't cover the changes that occur from those versions to Java 25. For information on migrating to Java 21 from those versions, check out our [migrate to Java 21 guide](./migrate-to-java-21.md).
+
+## Example Configuration
+
+The Java 25 migration recipe can be applied by including OpenRewrite's plugin to your project and including a dependency on [rewrite-migrate-java](https://github.com/openrewrite/rewrite-migrate-java):
+
+<Tabs groupId="projectType">
+<TabItem value="gradle" label="Gradle">
+
+```groovy title="build.gradle"
+  plugins {
+      id("java")
+      id("org.openrewrite.rewrite") version("{{VERSION_REWRITE_GRADLE_PLUGIN}}")
+  }
+  
+  rewrite {
+      activeRecipe("org.openrewrite.java.migrate.UpgradeToJava21")
+      setExportDatatables(true)
+  }
+  
+  repositories {
+      mavenCentral() // rewrite-spring is published to Maven Central
+  }
+  
+  dependencies {
+      rewrite(platform("org.openrewrite.recipe:rewrite-recipe-bom:latest.release"))
+      rewrite("org.openrewrite.recipe:rewrite-migrate-java")
+  
+      // Other project dependencies
+  }
+```
+
+</TabItem>
+
+<TabItem value="maven" label="Maven">
+```xml title="pom.xml"
```

Comment:
I think the tabs require a new line before any text block content.
```suggestion
<TabItem value="maven" label="Maven">

```xml title="pom.xml"
```